TrendMicroEmailSec/trendmicro_modules/trigger_email_sec.py

```diff
@@ -0,0 +1,256 @@
+import time
+import traceback
+import urllib.parse
+from functools import cached_property
+from threading import Event, Lock, Thread
+from typing import Generator
+
+import orjson
+from sekoia_automation.connector import Connector, DefaultConnectorConfiguration
+from sekoia_automation.storage import PersistentJSON
+
+from . import TrendMicroModule
+from .client import ApiClient, TrendMicroAuthentication
+from .helpers import iso8601_to_timestamp, unixtime_to_iso8601
+
+
+class TrendMicroConnectorConfiguration(DefaultConnectorConfiguration):
+    frequency: int = 60
+    batch_size: int = 500
+
+
+class TrendMicroWorker(Thread):
+    API_VERSION = "v1"
+
+    def __init__(self, connector: "TrendMicroEmailSecurityConnector", log_type: str):
+        super().__init__()
+
+        self.log_type = log_type
+        self.connector = connector
+
+        self.service_url = connector.module.configuration.service_url
+        if not self.service_url.startswith("https://"):
+            self.service_url = "https://%s" % self.service_url
+
+        self._stop_event = Event()
+
+    def stop(self):
+        self._stop_event.set()
+
+    @property
+    def running(self):
+        return not self._stop_event.is_set()
+
+    @property
+    def frequency(self):
+        return self.connector.configuration.frequency
+
+    @cached_property
+    def client(self) -> ApiClient:
+        auth = TrendMicroAuthentication(
+            username=self.connector.module.configuration.username,
+            api_key=self.connector.module.configuration.api_key,
+        )
```

Comment:
I think that [HTTPBasicAuth](https://docs.python-requests.org/en/latest/user/authentication/#basic-authentication) is more suitable.